orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -343,6 +343,9 @@ public void bootstrapNewNodes(List<GatewayConfig> allGatewayConfigs, Set<Node> t
     @Override
     public void initServiceRun(List<GatewayConfig> allGateway, Set<Node> allNodes, SaltConfig saltConfig, ExitCriteriaModel exitModel)
             throws CloudbreakOrchestratorException {
+        // TODO Question: Why does this get called with allNodes set to all nodes in the cluster,
```

Comment:
I think this method is used for both install and upscale, hence all nodes. Certainly it can be improved.

Reply:
I think because we modify the pillars when new nodes are added, and we have to apply those changes on the already existing node. One thing for sure is unbound configuration

Reply:
Thanks. Will look into this some more. Not making any changes in the initial patch, but I think this could be targeted for subsequent runs.
Am still curious about whether Unbound is still required - given IPA serves as the DNS.